quantum/painter/qp_draw_codec.c

```diff
@@ -32,7 +33,7 @@ bool qp_internal_decode_palette(painter_device_t device, uint32_t pixel_count, u
     const uint8_t pixels_per_byte  = 8 / bits_per_pixel;
     uint32_t      remaining_pixels = pixel_count; // don't try to derive from byte_count, we may not use an entire byte
     while (remaining_pixels > 0) {
-        uint8_t byteval = input_callback(input_arg);
```

Comment:
This should be `int16_t` not `uint16_t` right? Otherwise the `< 0` error check will never work.

Reply:
Looking a little deeper, yes, I agree, qp_internal_byte_input_callback is of type int16_t, so it looks like the use of uint16_t was just likely just a typo in the original qp code that I was basing my changes on. It is now corrected in both places to be int16_t.

Reply:
Thanks! I see that the bug is in `qp_draw_text.c` as well. Let me open a PR for that.

Reply:
Scratch the additional PR, it is fixed with this change :laughing: